orchestrate-migration.sh

```diff
@@ -0,0 +1,284 @@
+    if ls jira-keys-to-github-id-for-external-use*.txt 1> /dev/null 2>&1; then
+      cp jira-keys-to-github-id-for-external-use*.txt "$consolidated_mappings_dir/" 2>/dev/null || true
+    fi
+    mv jira-keys-to-github-id*.txt "$archive_dir/" 2>/dev/null || true
+  fi
+  
+  if [ -f "issue_jira_github_mapping.json" ]; then
+    mv issue_jira_github_mapping.json "$archive_dir/"
+  fi
+  
+  # Move watcher and remotelink files
+  if ls combined-*.txt 1> /dev/null 2>&1; then
+    mv combined-*.txt "$archive_dir/" 2>/dev/null || true
+  fi
+  
+  # Move jira comments file
+  if [ -f "jira-comments.txt" ]; then
+    mv jira-comments.txt "$archive_dir/"
+  fi
+  
+  log "✓ Archived outputs to $archive_dir"
+}
+
+# Function to cleanup temporary files
+cleanup_temp_files() {
+  log "Cleaning up temporary files"
+  
+  if [ -d "jira_output" ]; then
+    rm -rf jira_output/*
+    log "✓ Cleaned jira_output directory"
+  fi
+}
+
+# Function to process a single component
+process_component() {
+  local component=$1
+  local github_repo=$2
+  local github_owner="$JIRA_MIGRATION_GITHUB_NAME"
+  
+  log "=================================================="
+  log "Starting migration for component: $component"
+  log "Target: $github_owner/$github_repo"
+  log "=================================================="
+  
+  # Set component-specific environment variables
+  export JIRA_MIGRATION_GITHUB_REPO="$component"
+  export JIRA_MIGRATION_JQL_QUERY="project = ${JIRA_MIGRATION_JIRA_PROJECT_NAME} AND component in ($component) ORDER BY issuekey"
+  export JIRA_MIGRATION_FILE_PATHS="jira_output/combined.xml"
+  
+  # Set non-interactive mode variables
+  export JIRA_MIGRATION_START_FROM_INDEX=0
+  export JIRA_MIGRATION_DRY_RUN=false
+  export JIRA_MIGRATION_REFRESH_MAPPINGS=true
+  
+  log "JQL Query: $JIRA_MIGRATION_JQL_QUERY"
+  
+  # Step 1: Fetch issues from Jira
+  log ""
+  log "Step 1/9: Fetching issues from Jira..."
+  python3 ./fetch_issues.py
+  validate_file "jira_output/result-0.xml" "Jira issues XML"
+  
+  # Step 2: Concatenate XML results
+  log ""
+  log "Step 2/9: Concatenating XML results..."
+  ./concatenate-xml-results.sh
+  validate_file "jira_output/combined.xml" "Combined XML"
+  
+  # Clean up individual result files
+  rm -f jira_output/result-*.xml
+  log "✓ Cleaned up individual XML files"
+  
+  # Step 3: Retrieve watchers and remote links
+  log ""
+  log "Step 3/9: Retrieving watchers and remote links..."
+  ./retrieve-watchers-and-remotelinks.sh
+  validate_file "combined-remotelinks.txt" "Remote links"
+  
+  # Step 4: Enable GitHub Issues on repository
+  log ""
+  log "Step 4/9: Enabling GitHub Issues on repository..."
+  gh api -X PATCH "repos/$github_owner/$github_repo" --field has_issues=true || true
+  log "✓ GitHub Issues enabled (or already enabled)"
+  
+  # Step 5: Import issues to GitHub
+  log ""
+  log "Step 5/9: Importing issues to GitHub..."
+  python3 -u ./main.py
```

Comment:
if we don't pass unbuffered it only displays issues every 10 or so and looks like its hanging